src/components/CampaignForm.tsx

```diff
@@ -0,0 +1,204 @@
+import { Form } from 'react-final-form';
+import { useQuery } from 'react-query';
+import { useRouter } from 'next/router';
+import {  Autocomplete, TextField } from 'mui-rff';
+import { Avatar, Box, Button, MenuItem, Typography } from '@material-ui/core';
+import { Grid, GridSize } from '@material-ui/core';
+import { FormattedMessage as Msg, useIntl } from 'react-intl';
+import { useEffect, useState } from 'react';
+
+import getPeopleSearchResults from '../fetching/getPeopleSearchResults';
+import useDebounce from '../hooks/useDebounce';
+import { ZetkinCampaign, ZetkinPerson } from '../types/zetkin';
+
+interface CampaignFormProps {
+    campaign?: ZetkinCampaign;
+    onSubmit: (data: Record<string, unknown>) => void;
+    onCancel: () => void;
+}
+
+const CampaignForm = ({ onSubmit, onCancel, campaign }: CampaignFormProps): JSX.Element => {
+    const { orgId } = useRouter().query;
+    const intl = useIntl();
+
+    const [people, setPeople] = useState<ZetkinPerson[]>([]);
+    const [searchLabel, setSearchLabel] = useState(intl.formatMessage({ id: 'misc.formDialog.campaign.noOptions' }));
```

Comment:
Why state for this? It seems like derived state, where the data that actually decides which label should be visible are `isLoading` (from `useQuery()`) and `searchFieldValue.length`.